File: 06Adapter/Students/2018/DuniaWojciech/TwoWayAdapter/seabird.py
from iaircraft import IAirCraft
from seacraft import SeaCraft


class SeaBird(SeaCraft, IAirCraft):
    def __init__(self):
        self.airborne = False
        self.height = 0
        self.speed = 0

    @IAirCraft.airborne.getter
    def airborne(self):
        return self.height > 50

    def take_off(self):
        while not self.airborne:
            self.increase_revs()

    def increase_revs(self):
        super().increase_revs()
        if self.speed > 40:
            self.height += 100

File: 06Adapter/Students/2019/AliaksandrNovik/src/main/java/twoWayAdapter/Seabird.java
package twoWayAdapter;

public class Seabird extends Seacraft implements IAircraft {
	private int height = 0;

	@Override
	public void takeOff() {
		while (!airborne()) {
			increaseRevs();
		}
	}

	@Override
	public void increaseRevs() {
		super.increaseRevs();
		if (speed() > 40) {
			height += 100;
		}
	}

	public boolean airborne() {
		return speed() > 50;
	}

	public int height() {
		return height;
	}

}